src/Synths/IRMSynth.sol

```diff
@@ -26,15 +33,24 @@ contract IRMSynth is IIRM {
     IRMData internal irmStorage;
 
     error E_ZeroAddress();
+    error E_InvalidQuote();
 
-    constructor(address synth_, address referenceAsset_, address oracle_) {
+    constructor(address synth_, address referenceAsset_, address oracle_, uint256 targetQuoute_) {
         if (synth_ == address(0) || referenceAsset_ == address(0) || oracle_ == address(0)) {
             revert E_ZeroAddress();
         }
 
         synth = synth_;
         referenceAsset = referenceAsset_;
         oracle = IPriceOracle(oracle_);
+        targetQuote = targetQuoute_;
+        quoteAmount = 10 ** IERC20(synth_).decimals();
+
+        // Ensure the oracle works with the given synth and reference asset
+        uint256 testQuote = IPriceOracle(oracle_).getQuote(quoteAmount, synth_, referenceAsset_);
+        if (testQuote == 0) {
```

Comment:
I know we agreed to leave this on the call, but could we preempt any bug bounties with a comment here? Maybe like `// refusing to proceed with worthless asset` or something?